crates/polars-ops/src/chunked_array/list/product.rs

```diff
@@ -0,0 +1,76 @@
+use polars_core::export::num::NumCast;
+
+use super::*;
+
+fn product<T>(s: &Series) -> PolarsResult<T>
+where
+    T: NumCast,
+{
+    let prod = s.product()?.cast(&DataType::Float64)?;
+    Ok(T::from(prod.f64().unwrap().get(0).unwrap()).unwrap())
+}
+
+pub(super) fn product_with_nulls(ca: &ListChunked, inner_dtype: &DataType) -> PolarsResult<Series> {
```

Comment:
You could perhaps use this macro: https://docs.rs/polars-core/0.38.3/polars_core/macro.with_match_physical_numeric_type.html# to simplify the code

Reply:
Thanks for the comment. I see that the mapping (`DataType` -> physical type) in the macro is different  from the mapping this function uses. So I don't it can be used.